packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -74,14 +73,16 @@ export class SynthesisSession implements ISynthesisSession {
     return 'synthesize' in obj;
   }
 
-  public readonly store: ISessionStore;
+  public readonly outdir: string;
 
   private readonly artifacts: { [id: string]: cxapi.Artifact } = { };
-  private readonly buildSteps: { [id: string]: cxapi.BuildStep } = { };
   private _manifest?: cxapi.AssemblyManifest;
 
-  constructor(options: SynthesisOptions) {
-    this.store = options.store || new InMemoryStore();
+  constructor(options: SynthesisOptions = { }) {
+    this.outdir = options.outdir || fs.mkdtempSync(path.join(os.tmpdir(), 'cdk.out'));
```

Comment:
That directory isn't deleted by the program after it's done... This is filling up user's disks with junk in folders they never look at :/

Reply:
Correct, we can add this clean up later. It's mostly relevant for tests.
It is created under /tmp which is normally garbage collected by the system...